src/blueteam/Scout.java

```diff
@@ -1,18 +1,125 @@
 package blueteam;
 
-import battlecode.common.GameActionException;
-import battlecode.common.RobotController;
+import battlecode.common.*;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.Random;
+
 
 public class Scout extends Robot {
 
+	Direction dir;
+	Team enemy;
+	Random rand;
+	final int ROTATE_MIN = 30;
+	final int ROTATE_RANGE = 90;
+
 	Scout(RobotController rc) {
 		super(rc);
+		enemy = rc.getTeam().opponent();
+		dir = rc.getLocation().directionTo(rc.getInitialArchonLocations(enemy)[0]);
+		rand = new Random();
+	}
+
+	Optional<TreeInfo> findBulletTree(){
+		TreeInfo[] trees = rc.senseNearbyTrees(-1,Team.NEUTRAL);
+		if(trees.length > 0){
+			//res = Optional.of(trees[trees.length -1]);
+			for (TreeInfo tree:trees) {
+				if(tree.containedBullets > 0){
+					rc.setIndicatorDot(tree.getLocation(),250,0,0);
+					if(rc.canMove(tree.getLocation())) {
+						return Optional.of(tree);
+					}
+				}
+			}
+		}
+		return Optional.empty();
+	}
+
+	boolean shakeTheTree(TreeInfo tree) throws GameActionException{
+		if(rc.canShake(tree.getLocation())){
+			rc.shake(tree.getLocation());
+			return true;
+		}
+		return false;
+	}
+
+	Optional<RobotInfo> nearLumberjack(){
```

Comment:
we might want to push these to Robot too, probably make something like getNearestRobot(type, maxRadius) to make it more useful.